tests/pytests/test_empty.py

```diff
@@ -0,0 +1,655 @@
+        j = {
+            't': {}
+        }
+        js = json.dumps(j, separators=(',', ':'))
+        conn.execute_command('JSON.SET', 'j', '$', js)
+        cmd = f'FT.SEARCH {idx} isempty(@t)'.split(' ')
+        expected = [1, 'j', ['$', js]]
+        cmd_assert(env, cmd, expected)
+        """
+
+def testEmptyTag():
+    """Tests that empty values are indexed properly"""
+
+    env = Env(moduleArgs="DEFAULT_DIALECT 2")
+    conn = getConnectionByEnv(env)
+
+    def testEmptyTextHash(idx):
+        """Tests the indexing and querying of empty values for a TAG field of a
+        hash index"""
+
+        # Populate the db with a document that has an empty value for a TAG field
+        conn.execute_command('HSET', 'h1', 't', '')
+
+        # ------------------------- Simple retrieval ---------------------------
+        # Search for a single document, via its indexed empty value
+        cmd = f'FT.SEARCH {idx} isempty(@t)'.split(' ')
+        expected = [1, 'h1', ['t', '']]
+        cmd_assert(env, cmd, expected)
+
+        # Make sure the document is NOT returned when searching for a non-empty
+        # value
+        cmd = f'FT.SEARCH {idx} @t:foo'.split(' ')
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------ Negation ------------------------------
+        # Search for a negation of an empty value, make sure the document is NOT
+        # returned
+        cmd = f'FT.SEARCH {idx} -isempty(@t)'.split(' ')
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
+
+        # Search for a negation of a non-empty value, make sure the document is
+        # returned
+        cmd = f'FT.SEARCH {idx} -@t:{{foo}}'.split(' ')
+        expected = [1, 'h1', ['t', '']]
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------- Union --------------------------------
+        # Union of empty and non-empty values
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['isempty(@t) | @t:{foo}']
+        expected = [1, 'h1', ['t', '']]
+        cmd_assert(env, cmd, expected)
+
+        # ---------------------------- Intersection ----------------------------
+        # Intersection of empty and non-empty values
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['isempty(@t) @t:{foo}']
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------- Prefix -------------------------------
+        # We shouldn't get the document when searching for a prefix of "__empty"
+        cmd = f'FT.SEARCH {idx} @t:{{*pty}}'.split(' ')
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------- Suffix -------------------------------
+        # We shouldn't get the document when searching for a suffix of "__empty"
+        cmd = f'FT.SEARCH {idx} @t:{{__em*}}'.split(' ')
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
```

Comment:
Irrelevant now when using `isempty(@t)` instead of `__empty`?

Reply:
Answered above